Firebase.h

```diff
@@ -25,49 +25,108 @@
 #include <WiFiClientSecure.h>
 #include <ESP8266HTTPClient.h>
 
-// FirebaseError represents a Firebase API error with a code and a
-// message.
-class FirebaseError {
+//TODO(edcoyne) split these into multiple files.
+
+// Result from call to Firebase backend. ALWAYS check isError() before
+// expecting any data.
+class FirebaseResult {
  public:
-  operator bool() const { return _code < 0; }
-  int code() const { return _code; }
-  const String& message() const { return _message; }
-  void reset() { set(0, ""); }
-  void set(int code, const String& message) {
-    _code = code;
-    _message = message;
+  FirebaseResult(int status);
+  FirebaseResult(int status, const String& response);
+  FirebaseResult(const FirebaseResult& result);
+
+  // True if there was an error completeing call.
+  bool isError() const;
+  String errorMessage() const;
+
+  // True if http status code is 200(OK).
+  bool isOk() const;
+  // Message sent back from Firebase backend.
+  const String& response() const;
```

Comment:
So for direct key access of leaf value, we wouldn't even need json, because we can just atoi/atof.

But for non-leaf value, we have the choice of either a/ returning the raw body and letting the user decode or b/ wrap the decoding.

As discussed in #43, b/ is not recommended by `ArduinoJson` and Arduino IDE has poor (read no) library dependencies management.